corehq/util/context_processors.py

```diff
@@ -66,7 +66,7 @@ def current_url_name(request):
 
 
 def js_api_keys(request):
-    if not request.analytics_enabled:
+    if hasattr(request, 'couch_user') and not request.couch_user.analytics_enabled:
```

Comment:
I think if the request is mocked or otherwise instantiated outside of a normal request/response cycle, it won't go through middleware.  This was the specific problem: https://github.com/dimagi/commcare-hq/blob/25de9f0a17c911af9410c35cbf7ee9452f7918fc/corehq/apps/reports/views.py#L1131-L1135